proto/provenance/metadata/v1/query.proto

```diff
@@ -22,17 +22,22 @@ service Query {
 
   // Scope returns a specific scope by id
   rpc Scope(ScopeRequest) returns (ScopeResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_id}";
+    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_uuid}";
   }
 
   // GroupContext returns a specific group context within a scope (or all groups)
   rpc GroupContext(GroupContextRequest) returns (GroupContextResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_id}/group/{group_id}";
+    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_uuid}/group/{group_uuid}";
   }
 
-  // Record returns a collection of the records in a scope or a specific one by name
-  rpc Record(RecordRequest) returns (RecordResponse) {
-    option (google.api.http).get = "/provenance/metadata/v1/scope/{scope_id}/records/{name}";
+  // RecordByScopeUUID returns a collection of the records in a scope by scope uuid or a specific one by name
+  rpc RecordByScopeUUID(RecordByScopeUUIDRequest) returns (RecordByScopeUUIDResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/scope/uuid/{scope_uuid}/records/{name}";
+  }
+
+  // RecordByScopeID returns a collection of the records in a scope by scope bech32 id or a specific one by name
+  rpc RecordByScopeID(RecordByScopeIDRequest) returns (RecordByScopeIDResponse) {
+    option (google.api.http).get = "/provenance/metadata/v1/scope/id/{scope_id}/records/{name}";
   }
```

Comment:
These two functions should probably be plural since they return collections: `RecordsByScopeUUID` and `RecordByScopeID`. The message names should reflect that too.

Reply:
Agreed.